### PR TITLE
ASM-4072 use transport for razor resources

### PR DESCRIPTION
**Pull request by logankimmel**

### CODE REVIEW

**Comment by logankimmel** [comment]
marked as do-not merge even though it is probably safe to.  This PR is mostly just a placeholder for work in progress.

File: lib/puppet/type/rz_policy.rb [hunk 1]
```diff
@@ -75,6 +75,10 @@
     end
   end
 
+  newparam(:endpoint) do
```
Comment (gavin-scott) [inline lib/puppet/type/rz_policy.rb]: Is this not needed?

File: lib/puppet_x/puppetlabs/transport/razor.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  class Razor
+    attr_reader :name
+
+    def initialize(opts)
+      @name = opts[:name]
+      options  = opts[:options] || {}
+      @options = options.inject({}){|h, (k, v)| h[k.to_sym] = v; h}
+      @options[:host]     = opts[:server]
+      @options[:user]     = opts[:username]
+      @options[:password] = opts[:password]
```
Comment (gavin-scott) [inline lib/puppet_x/puppetlabs/transport/razor.rb]: @logankimmel is adding a client cert option for using client certs for the REST calls.

Comment (logankimmel) [inline lib/puppet_x/puppetlabs/transport/razor.rb]: the client_cert is an option that is passed in:
```
transport { 'logan' :
    server => '172.18.4.100',
    options => {
      'scheme' => 'https',
      'client_cert' => '/root/vmclient.pem',
    }
  }
```

**Comment by logankimmel** [comment]
@gavin-scott This is ready

**Pull request merged**